src/components/Explore/TokenTable/TokenTable.tsx

```diff
@@ -44,30 +52,76 @@ const LOADING_ROWS = Array(10)
   })
 
-function useFilteredTokens(topTokenAddresses: string[]) {
-  const filterString = useAtomValue<string>(filterStringAtom)
-  const favoriteTokens = useAtomValue<string[]>(favoritesAtom)
-  const showFavorites = useAtomValue<boolean>(showFavoritesAtom)
-  const shownTokens = showFavorites ? favoriteTokens : topTokenAddresses
+function useFilteredTokens(addresses: string[]) {
+  const filterString = useAtomValue(filterStringAtom)
+  const favoriteTokens = useAtomValue(favoritesAtom)
+  const showFavorites = useAtomValue(showFavoritesAtom)
+  const shownTokens = showFavorites ? favoriteTokens : addresses
   const allTokens = useAllTokens()
 
-  return useMemo(() => {
-    const filteredTokens = shownTokens.filter((tokenAddress) => {
-      const token = allTokens[tokenAddress]
-      const tokenName = token?.name ?? ''
-      const tokenSymbol = token?.symbol ?? ''
+  return useMemo(
+    () =>
+      shownTokens.filter((tokenAddress) => {
+        const token = allTokens[tokenAddress]
+        const tokenName = token?.name ?? ''
+        const tokenSymbol = token?.symbol ?? ''
+
+        if (!filterString) {
+          return true
+        }
+        const lowercaseFilterString = filterString.toLowerCase()
+        const addressIncludesFilterString = tokenAddress.toLowerCase().includes(lowercaseFilterString)
+        const nameIncludesFilterString = tokenName.toLowerCase().includes(lowercaseFilterString)
+        const symbolIncludesFilterString = tokenSymbol.toLowerCase().includes(lowercaseFilterString)
+        return nameIncludesFilterString || symbolIncludesFilterString || addressIncludesFilterString
+      }),
+    [allTokens, shownTokens, filterString]
+  )
+}
+
+function useSortedTokens(addresses: string[], tokenData: TokenData | null) {
+  const sortCategory = useAtomValue(sortCategoryAtom)
+  const sortDirection = useAtomValue(sortDirectionAtom)
+  const timePeriod = useAtomValue<TimePeriod>(filterTimeAtom)
 
-      if (!filterString) {
-        return true
-      }
-      const lowercaseFilterString = filterString.toLowerCase()
-      const addressIncludesFilterString = tokenAddress.toLowerCase().includes(lowercaseFilterString)
-      const nameIncludesFilterString = tokenName.toLowerCase().includes(lowercaseFilterString)
-      const symbolIncludesFilterString = tokenSymbol.toLowerCase().includes(lowercaseFilterString)
-      return nameIncludesFilterString || symbolIncludesFilterString || addressIncludesFilterString
-    })
-    return filteredTokens
-  }, [allTokens, shownTokens, filterString])
+  return useMemo(
+    () =>
+      addresses.sort((token1Address, token2Address) => {
+        if (!tokenData) {
+          return 0
+        }
+        const token1 = tokenData[token1Address] as any
+        const token2 = tokenData[token2Address] as any
+
+        if (!token1 || !token2 || !sortDirection || !sortCategory) {
+          return 0
+        }
+        if (sortCategory === Category.marketCap) {
```

Comment:
can you make a function called ```
function compare(sortDirection: SortDirection, a: any, b: any) {
  if (a === b) {
    return 0;
  } else if (a > b) {
    return sortDirection === SortDirection.decreasing ? -1 : 1
  } else {
    return sortDirection === SortDirection.decreasing ? 1 : -1
  }
}
```

then instead of doing branches for each sortCategory, you could just determine what `a` and `b` are based on the sort category.

```
let a;
let b;
switch (sortCategory) {
  Category.marketCap:
    a = token1.marketCap
    b = token2.marketCap
  ...
}
```

and finally it'd be
```
return compare(sortDirection, a, b)
```